contracts/system/TokensVesting.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "@solarity/solidity-lib/finance/vesting/Vesting.sol";
+
+import "./ContractKeys.sol";
+import "../interfaces/SignerOwnable.sol";
+import "../interfaces/IMintableToken.sol";
+import "../common/ContractRegistry.sol";
+
+contract TokensVesting is Vesting, SignerOwnable, ContractKeys {
+    ContractRegistry public contractsRegistry;
+
+    function initialize(address _signerGetterAddress, address _contractsRegistry) external initializer {
+        contractsRegistry = ContractRegistry(_contractsRegistry);
+
+        _setSignerGetter(_signerGetterAddress);
+    }
+
+    function createSchedule(BaseSchedule memory _baseSchedule) external onlySigner returns (uint256) {
```

Comment:
Who is the signer? The address that we pass in the initialize function?

Reply:
This contract used the same approach as other system contracts, so we keep the signerGetter contract, which on the core network is a DKG contract

Reply:
This is not related as well. System+operational contracts are not related to the token+vesting contracts.

Reply:
Got it, too. Then I'll make the vesting contract `Ownable`